src/systemctl/systemctl.c

```diff
@@ -4223,10 +4242,17 @@ static void print_status_info(
                 printf("     Loaded: %s%s%s (Reason: %s)\n",
                        on, strna(i->load_state), off, i->load_error);
         else if (path && !isempty(i->unit_file_state) && !isempty(i->unit_file_preset) &&
-                 !STR_IN_SET(i->unit_file_state, "generated", "transient"))
-                printf("     Loaded: %s%s%s (%s; %s; vendor preset: %s)\n",
-                       on, strna(i->load_state), off, path, i->unit_file_state, i->unit_file_preset);
-        else if (path && !isempty(i->unit_file_state))
+                 !STR_IN_SET(i->unit_file_state, "generated", "transient")) {
+                bool show_preset = show_preset_for_state(unit_file_state_from_string(i->unit_file_state));
+
+                printf("     Loaded: %s%s%s (%s; %s%s%s)\n",
+                       on, strna(i->load_state), off,
+                       path,
+                       i->unit_file_state,
+                       show_preset ? "; vendor preset: " : "",
+                       show_preset ? i->unit_file_preset : "");
+
+        } else if (path && !isempty(i->unit_file_state))
                 printf("     Loaded: %s%s%s (%s; %s)\n",
                        on, strna(i->load_state), off, path, i->unit_file_state);
```

Comment:
you are filtering the preset stuff twice now. the final branch was supposed to be the filtering branch...

i.e. the check for STR_IN_SET(i->unit_file_state, "generated", "transient") is to be subsumed by the show_preset_for_state() check...

Reply:
Duh.